inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape.cpp

```diff
@@ -182,6 +182,11 @@ bool DynamicToStaticShape::run_on_function(std::shared_ptr<ngraph::Function> fun
     // Operation-specific testing that needs to be performed in dynamic context before DSRs are introduced
     validateDynamicFunction(*function);
 
+    // Make sure all values are invalidated, we need it to correctly evaluate upper-bound
```

Comment:
Hi @jane-intel. Could you please take a look at this place? Without this invalidation some of our `StaticShape` operations will receive cached values, which were calculated before the model was converted to static. Is it okay to have such manual invalidation or it should have be done by the previous transformations? (In our case the previous transformation is `CommonOptimizations`)

Reply:
+ @GlebKazantaev

Reply:
Invalidation of partial values happens in the 'Function::validate_nodes_and_infer_types' method. More specifically -- it is a part of 'Node::revalidate_and_infer_type'. 
I see some custom validation earlier in the code -- I think we should change validate method to revalidate there and we could got rid of this manual value invalidation

Reply:
Unfortunately `revalidate_and_infer_types` does `validate_and_infer_types` right after values invalidation so the tensors again are filled with the dynamic values. It seems there is no other way rather than such kind of invalidation.

Reply:
Got it.

As we do not call Function::validate_nodes_and_infer_types after each StaticToShape transformations (it's okay as these transformations are special). 

Acknowledging that StaticToShape transformations should care about revalidation as it affects Node::validate_and_infer_types